File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        
        nums.sort()
        sol = []
        if(len(nums) < 3):
            return sol
        i = 0
        while(i < len(nums)-1 and nums[i] <= 0):
            j = i+1
            k = len(nums)-1
            while(j<k):
                sums = nums[i] + nums[j] + nums[k]
                if(sums == 0):
                    sol.append([nums[i], nums[j], nums[k]])
                    j += 1
                    k -= 1
                    while(nums[j] == nums[j-1] and j < len(nums)-1):
                        j += 1
                    while(nums[k] == nums[k+1] and k > 1):
                        k -= 1
                elif(sums < 0):
                    j += 1
                else:
                    k -= 1
            
            i += 1
            while(nums[i] == nums[i-1] and i < len(nums)-1):
                i += 1
        
        return sol

File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int last = 0;
        for (int i = 0; i < nums.length-last; i++) {
            if (nums[i] == val) {
                nums[i] = nums[nums.length-last-1];
                nums[nums.length-last-1] = val;
                last++;
            }
            if(nums[i] == val) i--;
        }
        return nums.length-last;
    }
}